### PR TITLE
CLI: Add flag to disable version checks

### PR DESCRIPTION
**Pull request by ndelangen**
Issue: #6194

## What I did
I made it so the version check only happens once in node, so it's less disruptive
I added a CLI flag that disables the feature entirely

## How to test

run an example with `--no-version-updates`, it should not do the version check
run an example without `--no-version-updates`, it should do the version check

### CODE REVIEW

File: lib/api/src/modules/versions.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import { fetch } from 'global';
+import { VERSIONCHECK } from 'global';
 import semver from 'semver';
-import { logger } from '@storybook/client-logger';
+import memoize from 'memoizerific';
 
+import { version } from 'punycode';
```
Comment (shilman) [inline lib/api/src/modules/versions.ts]: Unused (deepscan)

File: lib/api/src/tests/versions.test.js [hunk 2]
```diff
@@ -232,74 +158,20 @@ describe('versions API', () => {
...
 
     expect(api.getLatestVersion()).toMatchObject({
-      version: '4.0.0',
+      version: '5.2.3',
     });
   });
 
-  describe('versionUpdateAvailable', () => {
-    describe('stable current version', () => {
-      it('new latest version', async () => {
```
Comment (shilman) [inline lib/api/src/tests/versions.test.js]: There should be tests for these cases. Perhaps these need to be refactored to match the new structure?

Specifically the behavior is different depending on whether the user is running a prerelease or a stable version of storybook.

**Review by shilman — CHANGES_REQUESTED** [review]
Looks great! There are some deleted tests that need to get ported to the new structure.

File: lib/api/src/tests/versions.test.js (hunk 2 above)
Comment (ndelangen) [reply lib/api/src/tests/versions.test.js]: I changed these, into a different section in the testfile, should be good now

File: lib/api/src/modules/versions.ts (hunk 1 above)
Comment (ndelangen) [reply lib/api/src/modules/versions.ts]: removed

File: lib/api/src/tests/versions.test.js [hunk 3]
```diff
@@ -232,74 +162,82 @@ describe('versions API', () => {
...
 
-      it('new next version', async () => {
-        const store = createMockStore();
-        const { api, init, state: initialState } = initVersions({ store });
-        initialState.versions.current.version = '3.1.0-alpha.0';
-        store.setState(initialState);
+      expect(api.versionUpdateAvailable()).toEqual(true);
+    });
+
+    it('new major version', async () => {
```
Comment (shilman) [inline lib/api/src/tests/versions.test.js]: @ndelangen What about prerelease versions?

**Review by shilman — CHANGES_REQUESTED** [review]
Possibly one more case to handle?

**Comment by ndelangen** [comment]
added the test @shilman

File: lib/api/src/tests/versions.test.js [hunk 4]
```diff
@@ -239,5 +239,23 @@ describe('versions API', () => {
 
       expect(api.versionUpdateAvailable()).toEqual(true);
     });
+
+    it('new prerelease version', async () => {
```
Comment (shilman) [inline lib/api/src/tests/versions.test.js]: Will it work when the user is upgrading from one prerelease to another?

Comment (ndelangen) [reply lib/api/src/tests/versions.test.js]: added code & tests for this

**Review by shilman — APPROVED** [review]
BINGO!!! 💥

**Pull request merged**